compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -1114,8 +1119,8 @@ impl<'context> Elaborator<'context> {
     }
 
     fn elaborate_global(&mut self, global: UnresolvedGlobal) {
-        self.local_module = global.module_id;
-        self.file = global.file_id;
+        let old_module = std::mem::replace(&mut self.local_module, global.module_id);
+        let old_file = std::mem::replace(&mut self.file, global.file_id);
```

Comment:
This was the source of the error.
Most of the other changes with recovering `self.in_contract`, `self.file`, and `self.self_type` were me seeing if they'd fix anything. They did not but I figure those changes are good to have anyway.